### ISSUE TITLE
Show default values for positional arguments in `--help`

### ISSUE THREAD

**Issue by adosar** [omni-us/jsonargparse#689]
## 🚀 Feature request

Running `--help` in the CLI should print the default values of positional arguments.

### Motivation

User should be able to see all the default values that will be passed.

In the following example
```python
from jsonargparse import ArgumentParser

parser = ArgumentParser()
parser.add_argument('foo', default='a', type=str)
parser.add_argument('--bar', default=2, type=int)
cfg = parser.parse_args()
```
only the default value of the optional `bar` argument is shown:

```
usage: parser.py [-h] [--bar BAR] foo

positional arguments:
  foo         (required, type: str)

options:
  -h, --help  Show this help message and exit.
  --bar BAR   (type: int, default: 2)
```

### Pitch

```
usage: parser.py [-h] [--bar BAR] foo

positional arguments:
  foo         (required, type: str, default: a)

options:
  -h, --help  Show this help message and exit.
  --bar BAR   (type: int, default: 2)
```

**Comment by mauvilsa** [comment]
Thanks for proposing a feature!

This request is somewhat unexpected. Positionals are supposed to be required, as in, the user must provide a value. I tried giving a default to a positional with `argparse` and surprising it doesn't complain. Though, the default is useless, because the positional must always be provided, which means the default is always replaced.

Currently in `jsonargparse` not providing the positional succeeds, though, this is **not** an intended behavior. Just because right now this doesn't fail, it doesn't mean that after an internal change (e.g. to fix #427) the same behavior would be keep. I am tempted to fail adding the argument if a positional gets a default.

Anyway, the motivation is not clear. Why is it that you want to have a default for a positional? Is it just that you want the argument to be optional, but from command line be positional, not requiring a `--key`? In a sense this could be the same as #690.

**Comment by adosar** [comment]
Hi @mauvilsa ,

My motivation for the default was because it proceeded without failing if I don't specify a value for the positional argument. **Since this is not the intended behavior** and will be changed there is no reason for this feature
